print-babel.c

```diff
@@ -732,6 +803,13 @@ babel_print_v2(netdissect_options *ndo,
         }
         i += len + 2;
     }
+
+    /* Packet trailer */
+    if (i >= bodylen && (length >= (u_int)bodylen+4) && (length-bodylen-4 != 0)) {
```

Comment:
This should be done by putting the TLV parsing into a separate routine, called once for the body and once for the trailer.  (There should probably be a flag indicating whether it's the body or the trailer, so that TLVs not allowed to be in the trailer should be reported as such.)